src/Microsoft.OData.Core/UriParser/Resolver/UnqualifiedODataUriResolver.cs

```diff
@@ -46,28 +46,50 @@ public override IEnumerable<IEdmOperation> ResolveBoundOperations(IEdmModel mode
             {
                 return base.ResolveBoundOperations(model, identifier, bindingType);
             }
-
-            return FindAcrossModels<IEdmOperation>(model, identifier, this.EnableCaseInsensitive)
-                .Where(operation =>
-                    operation.IsBound
-                    && operation.Parameters.Any()
-                    && operation.HasEquivalentBindingType(bindingType));
+          
+            return GetBoundOperationsForModel(model, identifier, bindingType, this.EnableCaseInsensitive? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal);
         }
 
-        private static IEnumerable<T> FindAcrossModels<T>(IEdmModel model, String qualifiedName, bool caseInsensitive) where T : IEdmSchemaElement
+        private static IList<IEdmOperation> GetUnBoundOperationsForModel(IEdmModel model, string qualifiedName, StringComparison strComparison)
         {
-            Func<IEdmModel, IEnumerable<T>> finder = (refModel) =>
-                refModel.SchemaElements.OfType<T>()
-                .Where(e => string.Equals(qualifiedName, e.Name, caseInsensitive ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal));
+            IList<IEdmOperation> results = new List<IEdmOperation>();

```

Comment:
>IList<IEdmOperation> results = new List<IEdmOperation>(); [](http://example.com/codeflow?start=12&length=57)

rather than return a list, make the signature IEnumerable<IEdmOperation> and just yield on lines 61 and 71.